DNN Platform/DotNetNuke.DependencyInjection/Extensions/BuildUpExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.DependencyInjection.Extensions
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// Adds property injection extension methods.
+    /// </summary>
+    public static class BuildUpExtensions
```

Comment:
One possible thought here, not that this is a good thing...but a thought.  Should this be moved as an internal to the implementation in the action filter area to prevent mis-use?

Reply:
This is something I was thinking, but it does not work without making a few other changes. The Web API code is contained in the `DotNetNuke.Web` project, which means we need to add `[InternalsVisibleTo]` assembly declaration in the Dependency Injection project. 

I think this would be an acceptable solution

Reply:
Honestly, I believe that is a great solution for this, and I'm sure other things that we need to do, but only for internal benefit.  it would prevent us from just making everything public all the time

Reply:
This needs to be moved into `DotNetNuke.Web` because I don't want `DotNetNuke.DependencyInjection` to take a dependency on `System.Web`. 

The new plan will be keeping the exact same namespace but move it into the `DotNetNuke.Web` project. This is a common technique I've seen in various DI projects across .NET

Reply:
I'm going to resolve this conversation since I have moved the code. Please open a new one if you think there needs more discussion